tests/server/socksd.c

```diff
@@ -169,7 +169,7 @@ static unsigned char byteval(char *value)
 static unsigned short shortval(char *value)
 {
   unsigned long num = strtoul(value, NULL, 10);
-  return num & 0xffff;
+  return (short)num & 0xffff;
```

Comment:
I think this is curious. We build on *many* compilers and none of them seems to have a problem with this. On what basis do we need a typecast there?

Reply:
And if we need one, shouldn't be it `unsigned short` ?

Reply:
Agreed, it should've been `unsigned short`. Again, we're implicitly truncating a `long`, so the linter complained. With the cast at least its explicit (yeah yeah, I know it's just a two line func)